src/db_impl.cc

```diff
@@ -1137,7 +1137,14 @@ void TitanDBImpl::OnFlushCompleted(const FlushJobInfo& flush_job_info) {
         assert(false);
         delta = 0;
       }
+
+      // the metrics is counted in the table builder when flushing,
+      // so update it when updateing the live data size.
+      SubStats(stats_.get(), flush_job_info.cf_id,
```

Comment:
This is when we initially set `live_data_size` for the file. Where do we add stats before? I think we should not add stats before we do here.

Reply:
https://github.com/tikv/titan/blob/927ee60912d7b39085849899e5e67d765c8bef35/src/blob_storage.cc#L89-L98
When applying the edit, I think it's the right place to update the stat which is more aligned to sematic logic.
So I just handle the special cases instead of removing the code where we do before.